### DISCUSSION TITLE
Does workflow `on: pull_request, types: [opened]` only run when PR is mergeable?

### DISCUSSION THREAD

**Discussion by R2ZER0** [community/community#25984] [Actions]
<p>Hello,</p>
<p>Yesterday I was trying to run a workflow <code>on: pull_request</code> with <code>types: [opened, reopened]</code>.</p>
<p>I opened, closed, and re-opened the PR containing the workflow yaml several times and no workflow run was triggered. This PR was in a not-automatically-mergeable state.</p>
<p>Then I rebased the PR branch onto master, so that the PR was now automatically mergeable, and suddenly opening/closing/re-opening started to trigger the workflows.</p>
<p>So I’m wondering, is it expected or specified that <code>on: pull_request</code> workflows only run if the PR is mergeable? Or am I mistaken and something else is going on?</p>
<p>Thanks,<br>
Rikki</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/r2zer0">@R2ZER0</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>‘not-automatically-mergeable’ state means there’re some conflicts existing in the pull request.<br>
There’s a limitation that workflow is not triggered for ‘pull_request’ event if conflicts exist, you have to resolve it firstly, via web editor or command line tool on local machine.<br>
Rebase resolved the conflict, so the workflow is triggered.</p>
<p>Hope it’s clear and helpful!</p>

**Comment by R2ZER0** [comment]
<p>Hi,</p>
<p>Thanks for clarifying!</p>
<p>Does this limitation apply to <code>pull_request</code> webhooks too, or just running workflows?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/r2zer0">@R2ZER0</a>,</p>
<p>Checked on my side, the webhook will still send out the payload even when conflicts existing on pull request.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/e67128a9308fddd5313a0fd29604df83b8dd7533.png" data-download-href="/uploads/short-url/wSAg6Z8ztDShsbXvjcexUTpKRHR.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181094427-e445ac92-1e70-470d-9ee0-f1091a7706e6.png" alt="image" data-base62-sha1="wSAg6Z8ztDShsbXvjcexUTpKRHR" width="690" height="193" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e67128a9308fddd5313a0fd29604df83b8dd7533_2_690x193.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/e67128a9308fddd5313a0fd29604df83b8dd7533_2_1035x289.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/e/e67128a9308fddd5313a0fd29604df83b8dd7533_2_1380x386.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e67128a9308fddd5313a0fd29604df83b8dd7533_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1554×435 103 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Thanks</p>
